client/allocrunner/taskrunner/task_runner.go

```diff
@@ -505,12 +514,14 @@ func (tr *TaskRunner) Run() {
 	var result *drivers.ExitResult
 
 	tr.stateLock.RLock()
-	dead := tr.state.State == structs.TaskStateDead
+	restoredDead := tr.state.State == structs.TaskStateDead
```

Comment:
As I mentioned in https://github.com/hashicorp/nomad/pull/14127#discussion_r948582301, I'm not sure if the restore logic is 100% correct. It would be good to investigate this more.

Reply:
Commented a bit further down on the `restoredDead` check.

Also, I'm not sure I understand the changed name. Calling it `restoredDead` reads to me like it's been restored in the past from the dead state. But I think the code change means "it's _currently_ dead but eligible to be restored"? `deadAndCanBeRestored` is kind of wordy but maybe there's something better that's not coming to mind?

Reply:
Yeah, good point. What got me confused initially was that this variable was never updated, so it was only `true` when restoring a dead task. 

Since my initial implementation was incorrect there's now another value that needs to be checked here, so I reverted this one to `dead`.